Examples/Io/Csv/src/CsvTrackingGeometryWriter.cpp

```diff
@@ -76,76 +78,67 @@ void writeSurface(SurfaceWriter& writer, const Acts::Surface& surface,
   data.rot_zv = transform(2, 1);
   data.rot_zw = transform(2, 2);
 
-  // module thickness
-  if (surface.associatedDetectorElement()) {
-    const auto* detElement =
-        dynamic_cast<const Acts::IdentifiedDetectorElement*>(
-            surface.associatedDetectorElement());
-    if (detElement) {
-      data.module_t = detElement->thickness() / Acts::UnitConstants::mm;
-    }
-  }
+  std::array<float*, 7> dataBoundParameters = {
+      &data.p0, &data.p1, &data.p2, &data.p3, &data.p4, &data.p5, &data.p6};
 
-  // bounds and pitch (if available)
   const auto& bounds = surface.bounds();
-  const auto* planarBounds = dynamic_cast<const Acts::PlanarBounds*>(&bounds);
-  if (planarBounds) {
-    // extract limits from value store
-    auto boundValues = surface.bounds().values();
-    if (boundValues.size() == 2) {
-      data.module_minhu = boundValues[0] / Acts::UnitConstants::mm;
-      data.module_maxhu = boundValues[0] / Acts::UnitConstants::mm;
-      data.module_hv = boundValues[1] / Acts::UnitConstants::mm;
-    } else if (boundValues.size() == 3) {
-      data.module_minhu = boundValues[0] / Acts::UnitConstants::mm;
-      data.module_maxhu = boundValues[1] / Acts::UnitConstants::mm;
-      data.module_hv = boundValues[2] / Acts::UnitConstants::mm;
-    }
-    // get the pitch from the digitization module
-    const auto* detElement =
-        dynamic_cast<const Acts::IdentifiedDetectorElement*>(
-            surface.associatedDetectorElement());
-    if (detElement and detElement->digitizationModule()) {
-      auto dModule = detElement->digitizationModule();
-      // dynamic_cast to CartesianSegmentation
-      const auto* cSegmentation =
-          dynamic_cast<const Acts::CartesianSegmentation*>(
-              &(dModule->segmentation()));
-      if (cSegmentation) {
-        auto pitch = cSegmentation->pitch();
-        data.pitch_u = pitch.first / Acts::UnitConstants::mm;
-        data.pitch_u = pitch.second / Acts::UnitConstants::mm;
-      }
-    }
+  data.bounds_type = static_cast<int>(bounds.type());
+  auto boundValues = bounds.values();
+  for (size_t ipar = 0; ipar < boundValues.size(); ++ipar) {
+    (*dataBoundParameters[ipar]) = boundValues[ipar];
```

Comment:
You must check that the bound values do not exceed available parameters and you should set the unused variables to a placeholder value, either `0` or `NaN`.

Reply:
Oh, haven't addressed that, will do.

Reply:
Yes, that would be good.

Reply:
This addressed by an assert now.